pkg/snapshotmgr/snapshot_manager.go

```diff
@@ -542,95 +600,86 @@ func (this *SnapshotManager) CreateVolumeFromSnapshot(peID astrolabe.ProtectedEn
 		return
 	}
 	updatedID, err = astrolabe.NewProtectedEntityIDFromString(download.Status.VolumeID)
-	return
-}
 
-func (this *SnapshotManager) CreateVolumeFromSnapshotWithMetadata(peID astrolabe.ProtectedEntityID, metadata []byte, snapshotID string, backupRepositoryName string) (astrolabe.ProtectedEntityID, *v1api.Download, error) {
-	this.Infof("CreateVolumeFromSnapshotWithMetadata: Start creating restore for %s", peID.String())
-	// TODO(xyang): Will enable following code when astrolabe.Overwrite is ready
-	//config, err := rest.InClusterConfig()
-	//if err != nil {
-	//	this.WithError(err).Errorf("Failed to get k8s inClusterConfig")
-	//	return
-	//}
-	//pluginClient, err := plugin_clientset.NewForConfig(config)
-	//if err != nil {
-	//	this.WithError(err).Errorf("Failed to get k8s clientset with the given config: %v", config)
-	//	return
-	//}
+	// TODO(xyang): Watch for Download status and update CloneFromSnapshot status accordingly in Backupdriver
+	cloneFromSnap, err := pluginClient.BackupdriverV1().CloneFromSnapshots(cloneFromSnapshotNamespace).Get(cloneFromSnapshotName, metav1.GetOptions{})
+	if err != nil {
+		this.WithError(err).Errorf("CreateVolumeFromSnapshot: Failed to get CloneFromSnapshot %s/%s", cloneFromSnapshotNamespace, cloneFromSnapshotName)
+		return
+	}
+	clone := cloneFromSnap.DeepCopy()
+	// Since we wait until download is complete here,
+	// download.Status.Phase should be up to date.
+	clone.Status.Phase = backupdriverv1.ClonePhase(download.Status.Phase)
+	clone.Status.Message = download.Status.Message
+	apiGroup := ""
```

Comment:
Why is this an empty string?

Reply:
Core API group for resources such as PersistentVolumeClaim is an empty string.